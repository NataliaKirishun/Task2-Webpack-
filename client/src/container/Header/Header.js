import React, { Component } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Detail from '../../components/Detail/Detail';
import Search from '../../components/Search/Search';

class Header extends Component {
  state = {
    searchBy: 'title',
    term: '',
  };


  inputChangeHandler = (e) => {
    this.setState({
      term: e.target.value,
    });
  }

  toggleSearchBy = (e) => {
    this.setState({
      searchBy: e.target.value,
    });
  }

  formSubmitHandler = (e) => {
    const { searchBy, term } = this.state;
    e.preventDefault();
    this.props.searchFilmsHandler({
      searchBy,
      term: term.toLowerCase()});
    this.setState({
      term: '',
    });

  }

  render() {
    const activeFilm = this.props.activeFilm;
    return(
      <HeaderWrapper>
        {activeFilm
          ? (
            <Detail
              targetFilm={activeFilm}
              searchButtonHandler={this.props.searchButtonHandler} />
          )
          : (
            <Search
              searchBy={this.state.searchBy}
              inputChangeHandler={this.inputChangeHandler}
              toggleSearchBy={this.toggleSearchBy}
              formSubmitHandler={this.formSubmitHandler} />
          )
        }
      </HeaderWrapper>
    )
  }
}

export default Header;

Header.propTypes = {
  activeFilm: PropTypes.shape({}),
  searchFilmsHandler: PropTypes.func,
  searchButtonHandler: PropTypes.func,
};

const HeaderWrapper = styled.header`
  flex: 0 0 auto; 
`;