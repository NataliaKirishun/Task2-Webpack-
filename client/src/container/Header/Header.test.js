import React from 'react';
import { render } from 'enzyme';
import Header from './Header';

describe('Header Component Snapshot', () => {
    it('Header Component should render correctly with given props', () => {
        const activeFilm={
            "id":5924,
            "title":"Papillon",
            "tagline":"The greatest adventure of escape!",
            "vote_average":7.8,
            "vote_count":564,
            "release_date":"1973-12-13",
            "poster_path":"https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg",
            "overview":"A man befriends a fellow criminal as the two of them begin serving their sentence on a dreadful prison island, which inspires the man to plot his escape.",
            "budget":12000000,
            "revenue":53267000,
            "genres":["Crime","Drama"],
            "runtime":151
        };
        const searchFilmsHandler=jest.fn();
        const searchButtonHandler = jest.fn();
        const component = render(<Header activeFilm={activeFilm} searchFilmsHandler={searchFilmsHandler} searchButtonHandler={searchButtonHandler} />);
        expect(component).toMatchSnapshot();
    });
});

