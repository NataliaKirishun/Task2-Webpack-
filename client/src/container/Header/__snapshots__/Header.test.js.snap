// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component Snapshot Header Component should render correctly with given props 1`] = `
<header
  class="sc-kAzzGY bODEjt"
>
  <div
    class="sc-htpNat gfzgNE"
  >
    <div
      class="sc-bxivhb fCEMHd"
    >
      <div
        class="sc-ifAKCX dJXBli"
      >
        <a
          class="sc-bdVaJa gidgBK"
        >
          netflixroulette
        </a>
        <button
          class="sc-bwzfXH ftiveX"
        >
          Search
        </button>
      </div>
      <div
        class="sc-ifAKCX dJXBli"
      >
        <div
          class="sc-EHOje CbvPR"
        >
          <img
            class="sc-gzVnrw eVMzlz"
            src="https://image.tmdb.org/t/p/w500/xY7k2M9eElJnLSYwqY85Xon5z6m.jpg"
          />
        </div>
        <div
          class="sc-bZQynM eMGyob"
        >
          <h3
            class="sc-htoDjs jcOCnR"
          >
            Papillon
          </h3>
          <span
            class="sc-dnqmqq llFczy"
          >
            Crime, Drama
          </span>
          <div>
            <span
              class="sc-dnqmqq iexaMn"
            >
              1973
            </span>
            <span
              class="sc-dnqmqq iexaMn"
            >
              151min
            </span>
          </div>
          <span
            class="sc-dnqmqq llFczy"
          >
            A man befriends a fellow criminal as the two of them begin serving their sentence on a dreadful prison island, which inspires the man to plot his escape.
          </span>
        </div>
      </div>
    </div>
  </div>
</header>
`;
